modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -261,12 +264,38 @@ struct IEUnit {
 
     InferenceEngine::RemoteContext::Ptr rctx = nullptr;
 
+    std::shared_ptr<cv::gapi::wip::IPreprocEngine> preproc_engine_impl;
+
     // FIXME: Unlike loadNetwork case, importNetwork requires that preprocessing
     // should be passed as ExecutableNetwork::SetBlob method, so need to collect
     // and store this information at the graph compilation stage (outMeta) and use in runtime.
     using PreProcMap = std::unordered_map<std::string, IE::PreProcessInfo>;
     PreProcMap preproc_map;
 
+    // NEW FIXME: Need to aggregate getInputInfo & GetInputInfo from network
+    // into generic wrapper and invoke it at once in single place instead of
+    // analyzing ParamDesc::Kind::Load/Import every type when we need to get access
+    // for network info/
+    // In term of introducing custom VPP/VPL preprocessing functionality
+    // It was decided to use GFrameDesc as such aggregated network info with limitation
+    // that VPP/VPL produces cv::MediaFrame only. But it should be not considered as
+    // final solution
+    class InputFrameDesc {
```

Comment:
BTW, why don't you just leave:

```
std::unordered_map<std::string, cv::GFrameDesc> vpp_preproc_map;
```
and
```
cv::GFrameDesc configureVPPPreprocDesc(const IE::TensorDesc& desc) {
    const auto& dims = desc.getDims();
    return GFrameDesc{MediaFormat::NV12, cv::Size(dims[3], dims[2])};
}

for (auto &&it : ade::util::zip(ade::util::toRange(uu.params.input_names),
                                              ade::util::toRange(in_metas))) {
    ...
    if (preproc_engin_impl && cv::holds_alternative<cv::GFrameDesc>()) {
        vpp_preproc_map.emplace(input_name, configureVPPPreprocDesc(ii->getTensorDesc()));
    }
}
```

Reply:
I'd also accumulate this `map` and `vpp_preproc_engine` to some single place.

```
class VPPPreprocUnit {
    std::unordered_map<string, MediaFrame>         out_keep_alive_frames;
    std::shared_ptr<cv::gapi::wip::IPreprocEngine> preproc_engine_impl;
    std::unodered_map<string, GFrameDesc>          vpp_preproc_map
}
```

Reply:
1) it is intended to be a replacements for `IE::InputInfo:Ptr` and `IE::InputInfo::CPtr` without ugly if for load & import
2) it's not supposed to be a VPL/VPP specific `preproc_map` but input description for possible other preprocess engines

also, because current GIEbackend implementations requires refactoring anyway - i think it is not goof idea to make suggestions about such cosmetic changes before overall refactoring taken place